src/components/Game/GameArea.js

```diff
@@ -0,0 +1,138 @@
+import { useState } from 'react';
+
+import ActionButtons from 'components/Game/ActionButtons';
+import History from 'components/Game/History';
+import HealthMeter from "components/Game/HealthMeter";
+
+const PLAYER_ATTACK_MIN_PERCENTAGE = 1;
+const PLAYER_ATTACK_MAX_PERCENTAGE = 10;
+const MONSTER_ATTACK_MIN_PERCENTAGE = 1;
+const MONSTER_ATTACK_MAX_PERCENTAGE = 10;
+const SPECIAL_ATTACK_MIN_PERCENTAGE = 11;
+const SPECIAL_ATTACK_MAX_PERCENTAGE = 20;
+const SPECIAL_ATTACK_MIN_HEALTH = 90;
+const PLAYER_HEAL_PERCENTAGE = 10;
+const GAME_OVER_HEALTH = 0;
+
+const GameArea = (() => {
+  const gameState = {
+    player: 100,
+    enemy: 100,
+    started: false
+  };
+
+  const historyState = {
+    movesLog: []
+  };
+  
+  const randomNumber = ((min, max) => {
+    return Math.floor(Math.random() * (max - min)) + min;
+  });
+
+  const historyMessage = ((actionDoneBy, damage, actionType="attack") => {
+    if(actionType == "heal") {
+      return `Player heals for ${damage}`;
+    }
+    if(actionDoneBy == "player") {
+      return `Player hits monster for ${damage}`;
+    }
+    return `Monster hits player for ${damage}`;
+  });
+
+  const [currentGameState, updateGameState] = useState(gameState);
+  const [currentHistoryState, updateHistoryState] = useState(historyState);
```

Comment:
currentHistoryState => movesLog ?

Reply:
Didn't get this

Reply:
keeping history as an object is not required, therefore asked to change it to an array instead. And please change the name of history to something else as this is conflicting with the History object on window.
https://developer.mozilla.org/en-US/docs/Web/API/Window/history